src/types.ts

```diff
@@ -10,12 +10,26 @@ export type Logger = PinoLogger
 
 export type Primitive = string | boolean | number
 
-export interface ILightningWalletUser {
+export type WalletConstructorArgs = {
   // FIXME: Add a type for user here.
   user: unknown
   logger: Logger
 }
 
+export type UserWalletConstructorArgs = WalletConstructorArgs & {
+  config: UserWalletConfig
+}
+
+export type UserWalletConfig = {
+  dustThreshold: number
+  limits: {
```

Comment:
you can just use the `TransactionLimits` type here and call the functions in the code. No need to have _2_ layers of config wrapping